pinot-controller/src/test/java/com/linkedin/pinot/controller/api/restlet/resources/TableViewsTest.java

```diff
@@ -0,0 +1,206 @@
+    sendPostRequest(ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forTableCreate(),
+        request.toString());
+  }
+
+  @AfterClass
+  public void teardownTest()
+      throws Exception {
+    stopController();
+    stopZk();
+  }
+
+  @DataProvider(name = "stateProvider")
+  public Object[][] stateProvider() {
+    Object[][] configs = {
+        { "idealstate"},
+        { "externalview"}
+    };
+    return configs;
+  }
+
+  @Test(dataProvider = "stateProvider")
+  public void getOfflineTableState(String state)
+      throws IOException, JSONException {
+    String response = getState(OFFLINE_ONLY_TABLE, state, null);
+    TableViews.TableView tableView = toTableViews(response);
+    assertNotNull(tableView.offline);
+    assertNull(tableView.realtime);
+    assertEquals(tableView.offline.size(), 1);
+
+    for (Map.Entry<String, Map<String, String>> stringMapEntry : tableView.offline.entrySet()) {
+      assertTrue(stringMapEntry.getKey().startsWith("SimpleSegment"));
+      Map<String, String> serverMap = stringMapEntry.getValue();
+      assertEquals(serverMap.size(), 2);
+      for (Map.Entry<String, String> serverMapEntry : serverMap.entrySet()) {
+        assertTrue(serverMapEntry.getKey().startsWith("Server_"));
+        assertEquals(serverMapEntry.getValue(), "ONLINE");
+      }
+    }
+  }
+
+  @Test(dataProvider = "stateProvider")
+  public void testTableNotFound(String state)
+      throws IOException, JSONException {
+     ControllerRequestURLBuilder requestBuilder =
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL);
+    String url = requestBuilder.forTableView("UNKNOWN_TABLE", state, null);
+    HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
+    assertEquals(connection.getResponseCode(), 404);
+  }
+
+  @Test(dataProvider = "stateProvider")
+  public void testBadRequest(String state)
+      throws IOException {
+    ControllerRequestURLBuilder requestURLBuilder =
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL);
+    String url = requestURLBuilder.forTableView("UNKNOWN_TABLE", state, "no_such_type");
+    HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
+    assertEquals(connection.getResponseCode(), 400);
+  }
+
+  @Test(dataProvider = "stateProvider")
+  public void testGetState(String state)
+      throws IOException, JSONException {
+    String response = getState(TABLE_NAME, state, "realtime");
+    TableViews.TableView tableView = toTableViews(response);
+    assertNull(tableView.offline);
+    assertNotNull(tableView.realtime);
+    assertNotSame(tableView.realtime.size(), 0);
+
+    response = getState(TABLE_NAME, state, "offline");
+    tableView = toTableViews(response);
+    assertNull(tableView.realtime);
+    assertNotNull(tableView.offline);
+    // empty because we didn't add any segment
+    assertEquals(tableView.offline.size(), 0);
```

Comment:
I will not test for state changes as part of this PR. This code checks for representation of idealstate and externalview with nearly full coverage. Testing for state transitions is outside the scope unit testing for this code